foxai/explainer/computer_vision/algorithm/saliency.py

```diff
@@ -87,13 +87,19 @@ def calculate_features(
         Raises:
             RuntimeError: if attribution has shape (0).
         """
-        saliency = Saliency(forward_func=model)
+        grad_required = input_data.requires_grad
+        input_data.requires_grad_()
 
-        attributions: AttributionsType = saliency.attribute(
-            input_data,
-            target=pred_label_idx,
-            abs=abs_value,
-            additional_forward_args=additional_forward_args,
+        gradients = compute_gradients(
+            model, input_data, pred_label_idx, additional_forward_args
         )
+        if abs_value:
+            attributions = torch.abs(gradients)
+        else:
+            attributions = gradients
+
+        if not grad_required:
+            input_data.requires_grad_(False)
```

Comment:
If it is not time consuming, we might simplify:
```suggestion
        input_data.requires_grad_(grad_required)
```

generally it would make nice context manager:
```python
    with require_grad(input_data):
        ...
```

Reply:
Great idea, implemented.